hashexpr/src/lib.rs

```diff
@@ -82,16 +82,18 @@ impl PartialEq for Expr {
 
 impl Eq for Expr {}
 
-pub fn number_of_bytes(x: u64) -> u8 {
-  let mut n: u32 = 1;
-  let base: u64 = 256;
+#[must_use]
+pub const fn number_of_bytes(x: u64) -> u8 {
```

Comment:
Sure, `const fn` allows the function to be able to be called in a constant, and applied more optimizations to. "able to be computed at compile-time". It is generally good to make as many things `const` as possible because that opens up opportunities down the function dependency graph to be `const` as well.

https://rust-lang.github.io/rust-clippy/master/index.html#missing_const_for_fn